server/mdm/apple/commander.go

```diff
@@ -399,14 +401,27 @@ func (svc *MDMAppleCommander) SendNotifications(ctx context.Context, hostUUIDs [
 // APNSDeliveryError records an error and the associated host UUIDs in which it
 // occurred.
 type APNSDeliveryError struct {
-	FailedUUIDs []string
-	Err         error
+	errorsByUUID map[string]error
 }
 
 func (e *APNSDeliveryError) Error() string {
-	return fmt.Sprintf("APNS delivery failed with: %s, for UUIDs: %v", e.Err, e.FailedUUIDs)
+	var errStrings []string
+	for uuid, err := range e.errorsByUUID {
+		errStrings = append(errStrings, fmt.Sprintf("UUID: %s, Error: %v", uuid, err))
+	}
```

Comment:
Nit: might be nice to sort uuids alphabetically so the error message is deterministic (same for the `FailedUUIDs` method below).

Reply:
great idea, thank you! I pushed a fix.